src/coreclr/gc/gc.cpp

```diff
@@ -24,6 +24,14 @@
 #define USE_INTROSORT
 #endif
 
+#ifdef DACCESS_COMPILE
+#error this source file should not be compiled with DACCESS_COMPILE!
+#endif //DACCESS_COMPILE
+
+#if defined(DOUBLY_LINKED_FL) && !defined(BACKGROUND_GC)
+#error DOUBLY_LINKED_FL should only be used with BACKGROUND_GC
+#endif //DACCESS_COMPILE
```

Comment:
I don't think we need this - you are already explicitly only defining DOUBLY_LINKED_FL when BACKGROUND_GC is defined in gcpriv.h. checks like this are only for things that may be defined by things outside the GC, eg, DACCESS_COMPILE.